### DISCUSSION TITLE
How to enable TSO in QEMU ?

### DISCUSSION THREAD

**Discussion by freebrowser1** [utmapp/UTM#6850] [UTM Help (macOS)]
**Describe the issue**  
The version 4.62 allows to enable TSO for x64 VMs. Well I have a Ubuntu 22.04LTS VM (x64) and want to try this option.
But in the Settings screen of the VM I don't see any option to enable it. So how should I enable this ?

**Configuration**  
* UTM Version: 4.6.2
* macOS Version: 15.1
* Mac Chip : M1

**Crash log**  
N/A

**Debug log**  
N/A

**Comment by osy** [comment]
It's under QEMU

**Reply by freebrowser1** [reply]
> It's under QEMU

I cannot find it.
See here screenshot. Even the QEMU options don't show TSO.
<img width="848" alt="Screenshot 2024-11-29 at 22 31 07" src="https://github.com/user-attachments/assets/2d891a08-f7ff-418f-bd74-1ac99b747ac5">

**Reply by osy** [reply]
It only works for virtualized machines.

**Reply by freebrowser1** [reply]
So it does not work for emulated x64 machines, only for ARM (I have an M1 macbook) ?

**Reply by osy** [reply]
It doesn't make sense. TSO is a hardware feature. Emulation does not use the hardware. The use is for an ARM virtual machine to be able to emulate x86 memory access efficiently. When you emulate the entire x86_64 machine, the guest OS cannot even tell that it is on ARM and has no concept of "TSO".

**Reply by osy** [reply]
If you want to use TSO in emulation, you can run UTM on Rosetta but it will probably be way slower than "Force multicore" in native mode. However, it will never be incorrect.

**Reply by nolaforensix** [reply]
I think a concrete use case might ward off further confusion--I think on first read, many people believe it's a way to get emulated x86_64 VMs to run faster on Apple Silicon.  Everyone's hungry for that to be faster, of course.  Just a suggestion.
